fitlins/interfaces/nistats.py

```diff
@@ -184,37 +184,30 @@ def _run_interface(self, runtime):
         # Only keep files which match all entities for contrast
         stat_metadata = _flatten(self.inputs.stat_metadata)
         input_effects = _flatten(self.inputs.effect_maps)
-        # XXX nistats should begin supporting mixed effects models soon
-        # input_variances = _flatten(self.inputs.variance_maps)
-        input_variances = [None] * len(input_effects)
 
         filtered_effects = []
-        filtered_variances = []
         names = []
-        for m, eff, var in zip(stat_metadata, input_effects, input_variances):
+        for m, eff, var in zip(stat_metadata, input_effects):
             if _match(out_ents, m):
                 filtered_effects.append(eff)
-                filtered_variances.append(var)
                 names.append(m['contrast'])
 
-        for name, weights, contrast_type in prepare_contrasts(self.inputs.contrast_info, names):
-            # Need to add F-test support for intercept (more than one column)
-            # Currently only taking 0th column as intercept (t-test)
-            weights = weights[0]
-            effects = (np.array(filtered_effects)[weights != 0]).tolist()
-            _variances = (np.array(filtered_variances)[weights != 0]).tolist()
+        # Dummy code contrast of input effects
+        design_matrix = pd.get_dummies(names)
 
+        # Fit single model for all inputs
+        model.fit(filtered_effects, design_matrix=design_matrix)
+
+        for name, weights, contrast_type in prepare_contrasts(
+          self.inputs.contrast_info, design_matrix.columns.to_list()):
             contrast_metadata.append(
                 {'contrast': name,
                  'stat': contrast_type,
                  **out_ents})
 
-            design_matrix = pd.DataFrame(
-                {'intercept': weights[weights != 0]})
-
-            model.fit(effects, design_matrix=design_matrix)
-
             maps = model.compute_contrast(
+                con_val=weights,
+                contrast_type=contrast_type,
                 second_level_stat_type=contrast_type,
                 output_type='all')
```

Comment:
These aren't valid arguments:

https://github.com/nistats/nistats/blob/4b7adc71416504720fefcaf532ac16e7a13ae3f0/nistats/second_level_model.py#L413-L415

Reply:
Oh, I guess the args differ from the standalone function.

I wonder how you pass a 2d contrast matrix then? Anyways I'm out the rest of the week so I'll look at it Monday

Reply:
Oh, I guess the args differ from the standalone function.

I wonder how you pass a 2d contrast matrix then? Anyways I'm out the rest of the week so I'll look at it Monday